pulsar-common/src/main/java/org/apache/pulsar/common/topics/TopicList.java

```diff
@@ -83,15 +85,23 @@ public static Set<String> minus(Collection<String> list1, Collection<String> lis
         return s1;
     }
 
-    private static String removeTopicDomainScheme(String originalRegexp) {
+    @VisibleForTesting
+    static String removeTopicDomainScheme(String originalRegexp) {
         if (!originalRegexp.toString().contains(SCHEME_SEPARATOR)) {
             return originalRegexp;
         }
-        String removedTopicDomain = SCHEME_SEPARATOR_PATTERN.split(originalRegexp.toString())[1];
-        if (originalRegexp.contains("^")) {
-            return String.format("^%s", removedTopicDomain);
+        String[] parts = SCHEME_SEPARATOR_PATTERN.split(originalRegexp.toString());
+        String prefix = parts[0];
+        String removedTopicDomain = parts[1];
+        if (prefix.equals(TopicDomain.persistent.value()) || prefix.equals(TopicDomain.non_persistent.value())) {
+            prefix = "";
+        } else if (prefix.endsWith(TopicDomain.non_persistent.value())) {
+            prefix = prefix.substring(0, prefix.length() - TopicDomain.non_persistent.value().length());
+        } else if (prefix.endsWith(TopicDomain.persistent.value())){
+            prefix = prefix.substring(0, prefix.length() - TopicDomain.persistent.value().length());
```

Comment:
Do we have tests to cover this case?

Reply:
Yes, see: 
- https://github.com/apache/pulsar/pull/23014/files#diff-f68ba2fc789cf68f6cb7a9f588572d8743b98fd88b4fa72ac073855f551caeb8R1138-R1164
- https://github.com/apache/pulsar/pull/23014/files#diff-4104594a215409237c181ad71dede92abd0e4750b88d051fa5ba1327158f060dR114-R156